In [ ]:
#Loading dataset from sklearn -->

import pandas as pd
from sklearn.datasets import load_breast_cancer
d = load_breast_cancer()
df = pd.DataFrame(d['data'],columns=d['feature_names'])
df['target']=d['target']
print(df)

In [ ]:
#Printing  all keys of dataset in a list -->
print(list(df.keys()))

In [ ]:
#Intialising X with independent values -->

X= df.drop('target', axis='columns')
print(X)

In [ ]:
#Intialising y with dependent values -->

y=df['target']
print(y)

In [ ]:
# Importing tarining and testing algorithm -->

from sklearn.model_selection import train_test_split
X_train,X_test,y_train,y_test = train_test_split(X,y,test_size=0.3,random_state=10)

In [ ]:
# Importing KNeighbors Calssifier -->

from sklearn.neighbors import KNeighborsClassifier

In [ ]:
#Applying KNN with k=3 -->

model=KNeighborsClassifier(n_neighbors=3)
model.fit(X_train,y_train)
model.predict(X_test)
print(model.score(X_test,y_test))

0.935672514619883


In [ ]:
#Applying KNN with k=11 -->

model=KNeighborsClassifier(n_neighbors=11)
model.fit(X_train,y_train)
model.predict(X_test)
print(model.score(X_test,y_test))

0.9473684210526315


In [ ]:
#Applying KNN with k=31 -->

model=KNeighborsClassifier(n_neighbors=31)
model.fit(X_train,y_train)
model.predict(X_test)
print(model.score(X_test,y_test))

0.9298245614035088


In [ ]:
#Applying KNN with k=51 -->

model=KNeighborsClassifier(n_neighbors=51)
model.fit(X_train,y_train)
model.predict(X_test)
print(model.score(X_test,y_test))

0.9239766081871345


In [ ]:
#Applying KNN with k=101 -->

model=KNeighborsClassifier(n_neighbors=101)
model.fit(X_train,y_train)
model.predict(X_test)
print(model.score(X_test,y_test))

0.9064327485380117


 --> From the above results, value "K =11" gives the best result with Highest Accuracy of  ->  **94.7368%**